utils/avro-components-utils/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/SchemaRegistryProvider.scala

```diff
@@ -5,13 +5,14 @@ import io.confluent.kafka.schemaregistry.ParsedSchema
 import pl.touk.nussknacker.engine.avro.serialization.{KafkaAvroDeserializationSchemaFactory, KafkaAvroSerializationSchemaFactory}
 import pl.touk.nussknacker.engine.kafka.RecordFormatterFactory
 
-trait SchemaRegistryProvider extends Serializable with BaseSchemaRegistryProvider {
+//todo: rename and separate from BaseSchemaRegistryProvider
+trait SchemaRegistryProvider[T<:ParsedSchema] extends Serializable with BaseSchemaRegistryProvider {
```

Comment:
IMO The genesis of `SchemaRegistryProvider` name was "provider of implementation of mechanisms serializing/deserializing avro records".  You can imagine situations like:
- you have own, file based/configuration based schema registries (two our clients wanted such feature)
- you have other method of avro serailization - e.g. in confluent it is magic byte + 4bytes for schema id + payload, hermes uses schemaid in header, payload in body, I don't know how other schema registries works (e.g. hortonworks schema registry)

We can change the name to more suitable. Maybe: `SchemaBasedMessagesSerdeProvider`?
Good point that it should be separated from `BaseSchemaRegistryProvider` after adding this generic parameter.